spec/c14n_spec.rb

```diff
@@ -164,6 +164,10 @@
             canonicalize!(allow_query: :all).
             to_s.should eql('http://example.com/path?%ED=view')
         end
+
+        it 'does not error when there are bad things in query keys when allow_query isn\'t :all' do
+          BLURI('http://some.com/a/path?%E2').canonicalize!
```

Comment:
Would it be worth adding that it does not raise `EncodingError`? It would describe the bug that this commit fixes more accurately and would use an expectation like the other tests do.